File: P2/Fibonacci.py
# https://www.lintcode.com/problem/366/
class Fibonacci:
    def fibonacci(self, n: int) -> int:
        if n == 1:
            return 0
        if n == 2:
            return 1

        prev_prev = 0
        prev = 1
        result = 0
        for _ in range(3, n + 1):
            result = prev_prev + prev
            prev_prev = prev
            prev = result

        return result

File: old/java version(old)/important/Fibonacci.java
// https://www.lintcode.com/problem/366/
class Fibonacci {

  public int fibonacci(int n) {
    if (n == 1) {
      return 0;
    }
    if (n == 2) {
      return 1;
    }

    int prevPrev = 0;
    int prev = 1;
    int result = 0;
    for (int i = 3; i < n + 1; i++) {
      result = prevPrev + prev;
      prevPrev = prev;
      prev = result;
    }

    return result;
  }
}
